packages/api/endpoints/providers.js

```diff
@@ -55,25 +59,33 @@ async function get(req, res) {
  * @returns {Promise<Object>} the promise of express response object
  */
 async function post(req, res) {
-  const data = req.body;
-  const id = data.id;
-
-  const providerModel = new models.Provider();
-
   try {
-    // make sure the record doesn't exist
-    await providerModel.get({ id });
-    return res.boom.badReqest(`A record already exists for ${id}`);
-  } catch (e) {
-    if (e instanceof RecordDoesNotExist) {
-      const record = await providerModel.create(data);
-
-      if (inTestMode()) {
-        await addToLocalES(record, indexProvider);
+    const data = req.body;
+    const id = data.id;
+
+    const providerModel = new Provider();
+
+    try {
+      // make sure the record doesn't exist
+      await providerModel.get({ id });
+      return res.boom.badReqest(`A record already exists for ${id}`);
```

Comment:
```suggestion
      return res.boom.badRequest(`A record already exists for ${id}`);
```

Reply:
I know you didn't write that part, but if you could fix it that would be great and if you could put a test, that would be extra great.

Reply:
Wow, good catch